### PR TITLE
[PBE-0] support livestream

### PR DESCRIPTION
**Pull request by kanat**
### 🎯 Goal

Allow starting/joining livestreams from the Demo app

### 🎨 Showcase

| Scenario | Demo |
| -------- | ---- |
| Starting a Livestream |  <video src="https://github.com/user-attachments/assets/0a3320f2-7ddd-447d-b2cf-2b5d6422dea8"/>  |
| Joining a livestream with an ID | <video src="https://github.com/user-attachments/assets/dffc7852-c9a8-4590-bb56-f7122b4d1047"/> | 
| Joining a livestream with url | Just paste any call url with the `type` query parameter set to `livestream` |

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamVideoSwiftUI/Livestreaming/LivestreamPlayer.swift [hunk 1]
```diff
@@ -9,17 +9,21 @@ import SwiftUI
...
     @StateObject var state: CallState
     @StateObject var viewModel: LivestreamPlayerViewModel
-    
+
     public init(
         type: String,
         id: String,
         muted: Bool = false,
         showParticipantCount: Bool = true,
+        handleParticipationWithLifecycle: Bool = true,
```
Comment (martinmitrevski) [inline Sources/StreamVideoSwiftUI/Livestreaming/LivestreamPlayer.swift:25]: maybe we can simplify this, something like "joinImmediately" (or "joinAutomatically")?

File: Sources/StreamVideo/Utils/Sorting.swift [hunk 2]
```diff
@@ -34,13 +34,13 @@ public let defaultComparators: [StreamSortComparator<CallParticipant>] = [
...
 /// - `roles()`: Sorts participants based on their assigned roles.
 public let livestreamComparators: [StreamSortComparator<CallParticipant>] = [
-    ifInvisible(dominantSpeaker),
-    ifInvisible(isSpeaking),
-    ifInvisible(publishingVideo),
-    ifInvisible(publishingAudio),
-    roles(),
-    joinedAt,
-    userId
+    combineComparators([
```
Comment (martinmitrevski) [inline Sources/StreamVideo/Utils/Sorting.swift:37]: why do we need to change this?

**Review by martinmitrevski — COMMENTED** [review]
Looks good, just few small things. Also the call state has conflicts.

File: Sources/StreamVideo/Utils/Sorting.swift (hunk 2 above)
Comment (ipavlidakis) [reply Sources/StreamVideo/Utils/Sorting.swift:37]: This one is changed to match React sorting comparators as they seem to have been updated https://github.com/GetStream/stream-video-js/blob/3d115dd349fb0f4c57ff162dbaf829c2f96b45b9/packages/client/src/sorting/presets.ts#L93

File: Sources/StreamVideoSwiftUI/Livestreaming/LivestreamPlayer.swift [hunk 3]
```diff
@@ -5,21 +5,59 @@
...
+    ///   - showParticipantCount: Whether to show the count of participants. Defaults to `true`.
+    ///   - joinPolicy: The policy dictating how users join the livestream. Defaults to `.auto`.
+    ///   - showsLeaveCallButton: Whether to show a button to leave the call. Defaults to `false`.
+    ///   - onFullScreenStateChange: A callback for fullscreen state changes.
     public init(
         type: String,
         id: String,
         muted: Bool = false,
         showParticipantCount: Bool = true,
+        joinPolicy: JoinPolicy = .auto,
```
Comment (martinmitrevski) [inline Sources/StreamVideoSwiftUI/Livestreaming/LivestreamPlayer.swift:59]: Even better 👍

**Review by martinmitrevski — APPROVED** [review]
LGTM ✅

**Pull request merged**